homeassistant/config_entries.py

```diff
@@ -1672,11 +1673,20 @@ async def async_finish_flow(
 
 
 class OptionsFlow(data_entry_flow.FlowHandler):
-    """Base class for config option flows."""
+    """Base class for config options flows."""
 
     handler: str
 
 
+class OptionsFlowWithConfigEntry(OptionsFlow):
+    """Base class for options flows with config entry and options."""
+
+    def __init__(self, config_entry: ConfigEntry) -> None:
+        """Initialize options flow."""
+        self.config_entry = config_entry
+        self.options = deepcopy(dict(config_entry.options))
```

Comment:
Can we add a test which asserts the options are in fact deep copied?

Reply:
Test added